pyproject.toml

```diff
@@ -12,63 +12,29 @@ dependencies = [
     "httpx>=0.28.1",
     "graphviz>=0.20.3",
     "httpx>=0.28.1",
-    "matplotlib>=3.10.1",
     "networkx>=3.4.2",
-    "numpy<2.4",
     "pydantic>=2.11.3,<2.12.0",
     "pydot>=3.0.4",
-    "pyod==2.0.5",
-    "pythonping>=1.1.4",
     "pyyaml>=6.0.2",
     "requests>=2.32.3",
     "scapy>=2.6.1",
-    "sh>=2.2.2",
     "cyclopts>=3.22.5",
-    "whoosh-reloaded>=2.7.5",
     "boltons>=25.0.0",
-    "logparser>=0.8.4",
-    # start: drain3 dependencies
-    "jsonpickle",
-    # "cachetools",
-    # end: drain3 dependencies
-    "sparklines>=0.5.0",
     "pydantic-yaml>=1.5.1",
     "questionary>=2.1.0",
     "sentry-sdk>=2.29.1",
     "datadog-api-client[async]>=2.34.0",
-    "msgspec>=0.19.0",
     "kr8s>=0.20.7",
     "dotenv>=0.9.9",
     "pagerduty>=3.1.0",
     "fastapi[standard]>=0.116.1",
-    # start: dspy dependencies
-    "backoff>=2.2",
-    "joblib~=1.3",
-    "openai>=0.28.1",
-    "datasets>=2.14.6",
-    "regex>=2023.10.3",
-    "ujson>=5.8.0",
-    "tqdm>=4.66.1",
-    # "requests>=2.31.0",
-    "optuna>=3.4.0",
-    # "pydantic>=2.0",
-    "magicattr>=0.1.6",
-    "litellm>=1.64.0",
-    "diskcache>=5.6.0",
-    "json-repair>=0.30.0",
-    "tenacity>=8.2.3",
-    # "anyio",
-    "asyncer==0.0.8",
-    "cachetools>=5.5.0",
-    "cloudpickle>=3.0.0",
-    "rich>=13.7.1",
-    # end: dspy dependencies
     "uvicorn[standard]>=0.35.0",
     "pyngrok>=7.2.12",
     "pydantic-settings>=2.9.1",
     "mlflow>=3.1.4",
     "human-readable>=2.0.0",
-    "numba>=0.61.2",
+    "litellm>=1.75.5.post1",
+    "dspy>=3.0.0b4",
```

Comment:
This will get pinned to `==3.0.0b4` by the new build backend (#76), and in my local testing it was installed properly with no issues using both `uvx unpage` and `uv pip install unpage`.